File: CMDB/CMDB.UI.Specflow/Questions/Account/OpenTheAccountOverviewPage.cs
using Bright.ScreenPlay.Actors;
using CMDB.UI.Specflow.Abilities.Pages;
using Bright.ScreenPlay.Questions;
using CMDB.UI.Specflow.Abilities.Pages.AccountPages;
using OpenQA.Selenium;

namespace CMDB.UI.Specflow.Questions.Account
{
    public class OpenTheAccountOverviewPage : Question<AccountOverviewPage>
    {
        public override AccountOverviewPage PerformAs(IPerformer actor)
        {
            var page = actor.GetAbility<MainPage>();
            page.WaitUntilElmentVisableByXpath("//a[@id='Account']");
            page.ClickElementByXpath("//a[@id='Account']");
            page.ClickElementByXpath("//a[@id='Account5']");
            page.ClickElementByXpath("//a[@href='/Account']");
            page.WaitOnAddNew();
            AccountOverviewPage accountOverview= WebPageFactory.Create<AccountOverviewPage>(page.WebDriver);
            return accountOverview;
        }
    }
}

File: Java/playwright/src/test/java/be/hans/cmdb/Questions/Account/OpenTheAccountOverviewPage.java
package be.hans.cmdb.Questions.Account;

import be.brightest.ScreenPlay.Abilities.WebPageFactory;
import be.brightest.ScreenPlay.Actors.IActor;
import be.brightest.ScreenPlay.Question.Question;
import be.cmdb.pages.Account.AccountOverviewPage;
import be.cmdb.pages.MainPage;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.AriaRole;

/**
 * Question to open the Account Overview page.
 */
public class OpenTheAccountOverviewPage extends Question<AccountOverviewPage> {

    /**
     * Performs the action of opening the Account Overview page.
     * @param actor The actor performing the action
     * @return AccountOverviewPage
     */
    @Override
    public AccountOverviewPage performAs(IActor actor) {
        MainPage ability = (MainPage) actor.getAbility(MainPage.class);
        ability.getPage().getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Account")).click();
        ability.getPage().locator("#Account5").click();
        ability.getPage().getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("Overview")).click();
        ability.newButton().waitFor();
        return WebPageFactory.create(AccountOverviewPage.class, ability.getPage());
    }
}
